### ISSUE TITLE
Redundant codes

### ISSUE THREAD

**Issue by zhiyuanyou** [zhiyuanyou/DeQA-Score#7]
In the `softkl_loss()` function of the `MPLUGOwl2LlamaForCausalLM` class, there are some codes as follows.

```
        num_vision_tokens = logits.shape[1] - labels.shape[1]
        idx_level_logit = idx_level_label + num_vision_tokens - 1
        logits_level_ids = logits[
            torch.arange(batch_size), idx_level_logit
        ].contiguous()  # [B, V]
```

However, actually, the labels input to the `softkl_loss()` function have been padded in `prepare_inputs_labels_for_multimodal()` function. Therefore, the length of logits and labels will be the same. Thus, `num_vision_tokens` will always be 0.
